packages/@uppy/xhr-upload/types/index.d.ts

```diff
@@ -15,7 +15,7 @@ export interface XHRUploadOptions extends PluginOptions {
   timeout?: number
   responseUrlFieldName?: string
   endpoint: string
-  method?: 'GET' | 'POST' | 'PUT' | 'HEAD' | 'get' | 'post' | 'put' | 'head'
+  method?: 'GET' | 'POST' | 'PUT' | 'HEAD' | 'PATCH' | 'get' | 'post' | 'put' | 'head' | 'patch'
```

Comment:
IMO we should defined all the methods listed in https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods, plus all the lowercase version of the [standard ones](https://fetch.spec.whatwg.org/#concept-method-normalize), minus the [ones that would send a `SecurityError`](https://fetch.spec.whatwg.org/#forbidden-method). As discussed in the previous issue, lowercase `patch` is not valid, and only supported on 3.x for backward-compatibility.

```suggestion
  method?:
    | 'GET'
    | 'HEAD'
    | 'POST'
    | 'PUT'
    | 'DELETE'
    | 'OPTIONS'
    | 'PATCH'
    | 'delete'
    | 'get'
    | 'head'
    | 'options'
    | 'post'
    | 'put'
```
Could you also update the type in the source file please? Currently it is lacking quite a bunch of valid types: https://github.com/transloadit/uppy/blob/b5df7d05c6c1016b7895f2959dd34718e3960905/packages/%40uppy/xhr-upload/src/index.ts#L46